<launch> 
	<!-- rosaria node -->
	<include file="$(find pioneer_hallway)/launch/rosaria_pioneer_hallway.launch"/>
	
	<!-- start nodes for pioneer sensors -->
	<include file="$(find pioneer_hallway)/launch/pioneer_hallway_sensors.launch"/>
	
	<!-- move_base node -->
	<include file="$(find pioneer_hallway)/launch/2dnav_pioneer_hallway.launch"/>

	<!-- controller node -->
	<include file="$(find pioneer_test)/launch/pioneer_controller_spin_recover.launch"/>
	
	<!-- start gmapping -->
	<node pkg="gmapping" type="slam_gmapping" name="gmapping">
		<remap from="scan" to="RosAria/lms5XX_1_laserscan"/>
		<remap from="RosAria/lms5XX_1_laserscan" to="base_scan"/>
	</node>
	
	<!-- start rviz -->
	<include file="$(find pioneer_hallway)/launch/rviz_pioneer_hallway.launch"/>
</launch>
